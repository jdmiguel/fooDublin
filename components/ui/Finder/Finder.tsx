import React, { useState, useCallback } from 'react';
import { useDispatch } from 'react-redux';
import { useRouter } from 'next/router';

import { useWindowMeasurement } from '../../hooks/useWindowMeasurement';

import {
  StyledFinder,
  StyledAutocompleteMobile,
  StyledAutocomplete,
  StyledDropdownsWrapper,
  StyledDropdown,
  StyledSpacer,
  StyledButton,
} from './styles';

import { setRelatedRestaurants } from '@/store/redux/actions';

import {
  DUBLIN_ID,
  LOCATIONS,
  CUISINES,
  THUMB_GENERIC_SRC,
  MIN_RESTAURANTS_LIST,
} from '@/store/statics';
import {
  getFormattedUrlText,
  getCurrentRelatedRestaurants,
} from '@/helpers/utils';

import { getRestaurants } from '@/services/index';

import { EntityType, Restaurant, RawRestaurant } from '../../pages/types';

type FinderProps = {
  className?: string;
};

export const Finder: React.FC<FinderProps> = ({ className }) => {
  const [suggestions, setSuggestions] = useState<Restaurant[]>();
  const [isAutocompleteLoading, setIsAutocompleteLoading] = useState(false);
  const [isButtonLoading, setIsButtonLoading] = useState(false);
  const [currentLocationPath, setCurrentLocationPath] = useState('dublin');
  const [currentCuisinePath, setCurrentCuisinePath] = useState('any-food');

  const dispatch = useDispatch();

  const { width } = useWindowMeasurement();
  const isMobile = width < 768;

  const router = useRouter();

  const fetchSuggestions = useCallback(
    async (search: string) => {
      setIsAutocompleteLoading(true);

      const { rawRestaurants, status } = await getRestaurants({
        entity_id: DUBLIN_ID,
        entity_type: EntityType.CITY,
        cuisines: 0,
        q: search,
      });

      if (status === 200) {
        const formattedRestaurants = rawRestaurants.map(
          (rawRestaurant: RawRestaurant) => ({
            id: rawRestaurant.restaurant.id,
            imgSrc: rawRestaurant.restaurant.thumb || THUMB_GENERIC_SRC,
            title: rawRestaurant.restaurant.name,
            content: rawRestaurant.restaurant.location.locality,
            route: '/detail/[id]/[name]',
            asRoute: `/detail/${
              rawRestaurant.restaurant.id
            }/${getFormattedUrlText(rawRestaurant.restaurant.name, true)}`,
          }),
        );

        setSuggestions(formattedRestaurants);
        setIsAutocompleteLoading(false);
      }
    },
    [setIsAutocompleteLoading, setSuggestions],
  );

  const selectSuggestion = useCallback(
    (id: number, name: string) => {
      const path = getFormattedUrlText(name, true);

      if (
        Array.isArray(suggestions) &&
        suggestions.length > MIN_RESTAURANTS_LIST
      ) {
        const currentRelatedRestaurants = getCurrentRelatedRestaurants(
          suggestions,
          id,
        );

        dispatch(setRelatedRestaurants(currentRelatedRestaurants));
      }

      setIsButtonLoading(true);
      router.push('/detail/[id]/[name]', `/detail/${id}/${path}`);
    },
    [suggestions],
  );

  const handleButtonClick = () => {
    if (!isButtonLoading) {
      setIsButtonLoading(true);
      router.push(
        '/search/[location]/[cuisine]',
        `/search/${currentLocationPath}/${currentCuisinePath}`,
      );
    }
  };

  return (
    <StyledFinder className={className}>
      {isMobile ? (
        <StyledAutocompleteMobile
          suggestions={suggestions || []}
          fetchSuggestions={fetchSuggestions}
          selectSuggestion={selectSuggestion}
          disabled={isButtonLoading}
          loading={isAutocompleteLoading}
          hasSearchIcon={true}
        />
      ) : (
        <StyledAutocomplete
          suggestions={suggestions || []}
          fetchSuggestions={fetchSuggestions}
          selectSuggestion={selectSuggestion}
          disabled={isButtonLoading}
          loading={isAutocompleteLoading}
          hasSearchIcon={true}
        />
      )}
      <StyledSpacer />
      <StyledDropdownsWrapper>
        <StyledDropdown
          icon="near_me"
          labelTxt="Select any location"
          list={LOCATIONS}
          disabled={isButtonLoading}
          onSelect={(path: string) => setCurrentLocationPath(path)}
          onClear={() => setCurrentLocationPath('dublin')}
        />
        <StyledDropdown
          icon="restaurant"
          labelTxt="Select any cuisine"
          list={CUISINES}
          disabled={isButtonLoading}
          onSelect={(path: string) => setCurrentCuisinePath(path)}
          onClear={() => setCurrentCuisinePath('any-food')}
        />
      </StyledDropdownsWrapper>
      <StyledButton loading={isButtonLoading} onClick={handleButtonClick}>
        Search
      </StyledButton>
    </StyledFinder>
  );
};
