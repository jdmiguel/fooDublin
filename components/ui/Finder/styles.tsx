import styled from 'styled-components';

export const StyledFinder = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  @media only screen and (min-width: 768px) {
    display: flex;
    flex-direction: row;
  }
  @media only screen and (min-width: 1200px) {
    max-width: 1200px;
  }
`;

export const StyledAutocompleteMobileWrapper = styled.div`
  width: 100%;
  margin: 0 0 25px;
`;

export const StyledAutocompleteWrapper = styled.div`
  flex: 1;
  margin: 0 0 25px;
  @media only screen and (min-width: 768px) {
    flex: 0.5;
    margin: 0 10px 0 0;
  }
  @media only screen and (min-width: 992px) {
    flex: 1;
    margin: 0;
  }
`;

export const StyledSpacer = styled.span`
  background-color: ${({ theme }) => theme.palette.DARK_SOFT};
  width: 100%;
  height: 1px;
  margin: 0 0 25px 0;
  @media only screen and (min-width: 768px) {
    display: none;
  }
  @media only screen and (min-width: 992px) {
    display: block;
    width: 1px;
    height: 90%;
    margin: 0 20px 0;
  }
`;

export const StyledDropdownsWrapper = styled.div`
  flex: 1;
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin-bottom: 22px;
  &:first-child {
    margin-right: 10px;
  }
  @media only screen and (min-width: 768px) {
    flex-direction: row;
    justify-content: space-around;
    margin-bottom: 0;
    &:first-child {
      margin-right: 0;
    }
  }
`;

export const StyledDropdownWrapper = styled.div`
  width: 100%;
  margin: 0 10px 15px 0;
  @media only screen and (min-width: 768px) {
    margin: 0 10px 0 0;
  }
`;

export const StyledButtonWrapper = styled.div`
  flex: 1;
  margin-bottom: 10px;
  @media only screen and (min-width: 768px) {
    flex: 0;
    margin-bottom: 0;
  }
`;
