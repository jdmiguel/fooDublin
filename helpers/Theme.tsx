import React from 'react';
import { DefaultTheme, ThemeProvider } from 'styled-components';

export const theme: DefaultTheme = {
  palette: {
    PRIMARY: '#EC7E00',
    PRIMARY_DARK: '#D48C24',
    PRIMARY_MEDIUM: '#FBA52B',
    PRIMARY_LIGHT: '#FDF3E5',
    SECONDARY: '#660000',
    LIGHT_MAX: '#FFFFFF',
    LIGHT_MEDIUM: '#F3F3F3',
    LIGHT_MIN: '#E5EAED',
    LIGHT_SOFT: '#D8D8D8',
    DARK_MAX: '#222C3C',
    DARK_MEDIUM: '#585858',
    DARK_MIN: '#89959B',
    DARK_SOFT: '#A7A7A7',
  },
};

export const renderWithTheme = (component: any) => (
  <ThemeProvider theme={theme}>{component}</ThemeProvider>
);
