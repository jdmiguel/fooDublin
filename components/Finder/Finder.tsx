import React, { useState, useCallback, Dispatch } from 'react';
import { useRouter } from 'next/router';
import styled from 'styled-components';

import { getFormattedUrlText } from '../../helpers/utils';
import { THUMB_GENERIC_SRC } from '../../helpers/staticData';
import { Suggestion } from '../../helpers/types';

import useWindowMeasures from '../hooks/useWindowMeasures';

import { getRestaurantsData } from '../../services';

import AutocompleteMobile from '../core/Autocomplete/AutocompleteMobile';
import Autocomplete from '../core/Autocomplete/Autocomplete';
import Dropdown from '../core/Dropdown/Dropdown';
import Button from '../core/Button/Button';

import {
  DUBLIN_ID,
  DEFAULT_SUGGESTIONS,
  LOCATIONS,
  CUISINES,
} from '../../helpers/staticData';

type FinderProps = {
  className?: string;
};

const StyledFinder = styled.div`
  width: 85%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  @media only screen and (min-width: 470px) {
    width: 70%;
  }
  @media only screen and (min-width: 768px) {
    width: 60%;
  }
  @media only screen and (min-width: 992px) {
    width: 58rem;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
  }
  @media only screen and (min-width: 1250px) {
    width: 71rem;
  }
`;

const StyledAutocompleteMobile = styled(AutocompleteMobile)`
  width: 100%;
  margin: 0 0 25px;
`;

const StyledAutocomplete = styled(Autocomplete)`
  margin: 0 0 25px;
  @media only screen and (min-width: 992px) {
    margin: 0 2% 0 0;
    width: 39%;
  }
  @media only screen and (min-width: 1200px) {
    width: 44%;
  }
`;

const StyledSpacer = styled.span`
  background-color: ${(props) => props.theme.palette.DARK_SOFT};
  width: 100%;
  height: 1px;
  margin: 0 0 25px 0;
  max-width: 550px;
  @media only screen and (min-width: 992px) {
    width: 1px;
    height: 90%;
    margin: 0 2% 0 0;
  }
`;

const StyledDropdownsWrapper = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin-bottom: 22px;
  &:first-child {
    margin-right: 2%;
  }
  @media only screen and (min-width: 768px) {
    flex-direction: row;
  }
  @media only screen and (min-width: 992px) {
    width: 47%;
    justify-content: space-around;
    margin-bottom: 0;
    &:first-child {
      margin-right: 0;
    }
  }
`;

const StyledDropdown = styled(Dropdown)`
  width: 100%;
  &:first-of-type {
    margin-right: 2%;
    margin-bottom: 15px;
  }
  @media only screen and (min-width: 768px) {
    max-width: 270px;
    &:first-of-type {
      margin-bottom: 0;
    }
  }
  @media only screen and (min-width: 992px) {
    margin-right: 2%;
    width: 48%;
  }
`;

const StyledButton = styled(Button)`
  width: 100%;
  margin-bottom: 10px;
  @media only screen and (min-width: 768px) {
    width: 300px;
    margin-bottom: 0;
  }
  @media only screen and (min-width: 992px) {
    width: 10%;
  }
`;

const Finder: React.FC<FinderProps> = ({ className }) => {
  const [suggestions, setSuggestions]: [
    Suggestion[],
    Dispatch<Suggestion[]>,
  ] = useState(DEFAULT_SUGGESTIONS);
  const [isAutocompleteLoading, setIsAutocompleteLoading] = useState(false);
  const [isButtonLoading, setIsButtonLoading] = useState(false);
  const [currentLocationPath, setCurrentLocationPath] = useState('dublin');
  const [currentCuisinePath, setCurrentCuisinePath] = useState('any-food');

  const { width } = useWindowMeasures();
  const isMobile = width < 768;

  const router = useRouter();

  const fetchSuggestions = useCallback(
    async (search: string) => {
      setIsAutocompleteLoading(true);

      const response = await getRestaurantsData(DUBLIN_ID, 'city', 0, search);
      const restaurants = response.restaurants.map((restaurant: any) => ({
        id: restaurant.restaurant.id,
        imgSrc: restaurant.restaurant.thumb || THUMB_GENERIC_SRC,
        firstText: restaurant.restaurant.name,
        secondText: restaurant.restaurant.location.locality,
      }));

      setSuggestions(restaurants);
      setIsAutocompleteLoading(false);
    },
    [setIsAutocompleteLoading, setSuggestions],
  );

  const selectSuggestion = (name: string) => {
    const path = getFormattedUrlText(name, true);
    setIsButtonLoading(true);

    router
      .push('/detail/[name]', `/detail/${path}`)
      .then(() => window.scrollTo(0, 0));
  };

  const handleButtonClick = () => {
    if (!isButtonLoading) {
      setIsButtonLoading(true);
      router
        .push(
          '/search/[location]/[cuisine]',
          `/search/${currentLocationPath}/${currentCuisinePath}`,
        )
        .then(() => window.scrollTo(0, 0));
    }
  };

  return (
    <StyledFinder className={className}>
      {isMobile ? (
        <StyledAutocompleteMobile
          suggestions={suggestions}
          fetchSuggestions={fetchSuggestions}
          selectSuggestion={selectSuggestion}
          loading={isAutocompleteLoading}
          hasSearchIcon={true}
        />
      ) : (
        <StyledAutocomplete
          suggestions={suggestions}
          fetchSuggestions={fetchSuggestions}
          selectSuggestion={selectSuggestion}
          loading={isAutocompleteLoading}
          hasSearchIcon={true}
        />
      )}
      <StyledSpacer />
      <StyledDropdownsWrapper>
        <StyledDropdown
          icon="near_me"
          labelTxt="Select any location"
          list={LOCATIONS}
          onSelect={(path: string) => setCurrentLocationPath(path)}
          onClear={() => setCurrentLocationPath('dublin')}
        />
        <StyledDropdown
          icon="restaurant"
          labelTxt="Select any cuisine"
          list={CUISINES}
          onSelect={(path: string) => setCurrentCuisinePath(path)}
          onClear={() => setCurrentCuisinePath('any-food')}
        />
      </StyledDropdownsWrapper>
      <StyledButton loading={isButtonLoading} onClick={handleButtonClick}>
        Search
      </StyledButton>
    </StyledFinder>
  );
};

export default Finder;
