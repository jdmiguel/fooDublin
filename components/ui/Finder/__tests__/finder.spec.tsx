/**
 * @jest-environment jsdom
 */

import React from 'react';
import { Provider } from 'react-redux';
import { render, fireEvent } from '@testing-library/react';
import 'jest-styled-components';

import { createTestStore } from '../../../../helpers/utils';

import { Finder } from '../Finder';

import { renderWithTheme } from '../../../../helpers/Theme';

describe('Component: Finder', () => {
  let store;

  beforeEach(() => {
    store = createTestStore();
  });

  it('should render', () => {
    const { container } = render(
      renderWithTheme(
        <Provider store={store}>
          <Finder onNavigation={() => {}} />
        </Provider>,
      ),
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should call function on button click', () => {
    const handleClick = jest.fn();
    const { getByText } = render(
      renderWithTheme(
        <Provider store={store}>
          <Finder onNavigation={handleClick} />
        </Provider>,
      ),
    );

    fireEvent.click(getByText('Search'));

    expect(handleClick).toHaveBeenCalled();
  });
});
