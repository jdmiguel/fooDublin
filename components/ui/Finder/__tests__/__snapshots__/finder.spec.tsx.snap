// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Finder should render 1`] = `
.c3 {
  width: 100%;
  border-radius: 4px;
  padding: 0 20px;
  background: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 i {
  font-size: 1.2rem;
  color: #A7A7A7;
  margin-right: 10px;
}

.c5 {
  width: 100%;
  height: 55px;
  padding: 10px 0;
  line-height: 20px;
  border: none;
  outline: none;
  color: #2f3235;
  background: transparent;
  cursor: pointer;
  outline: none;
  -webkit-tap-highlight-color: transparent;
}

.c16 {
  color: #585858;
  font-size: 1.2rem;
  font-weight: 500;
}

.c1 {
  width: 100%;
  max-width: 550px;
  height: 55px;
  position: relative;
  cursor: pointer;
}

.c4 {
  background-color: #F3F3F3;
  -webkit-transition: background-color 0.2s ease-out;
  transition: background-color 0.2s ease-out;
}

.c4:hover {
  background-color: #FDF3E5;
}

.c6 {
  visibility: hidden;
  opacity: 0;
  -webkit-transform: translateY(10px);
  -ms-transform: translateY(10px);
  transform: translateY(10px);
  -webkit-transition: opacity 0.2s ease 0s,-webkit-transform 0.2s ease 0s;
  -webkit-transition: opacity 0.2s ease 0s,transform 0.2s ease 0s;
  transition: opacity 0.2s ease 0s,transform 0.2s ease 0s;
  position: absolute;
  box-sizing: border-box;
  background: #F3F3F3;
  z-index: 2;
  top: 55px;
  left: 0;
  padding: 0;
  width: 100%;
  height: auto;
  max-height: 400px;
  overflow: auto;
  box-shadow: 0 2px 2px rgba(0,0,0,0.35);
  border: 1px solid #F3F3F3;
  border-radius: 0 0 4px 4px;
  border-bottom-right-radius: 4px;
  border-bottom-left-radius: 4px;
  border-top: 0;
}

.c7 {
  width: 100%;
  -webkit-tap-highlight-color: transparent;
}

.c10 {
  width: 100%;
  max-width: 550px;
  height: 100%;
  position: initial;
}

.c12 {
  position: relative;
  border-radius: 4px;
  background-color: #F3F3F3;
}

.c13 {
  width: 100%;
  height: 55px;
  background-color: #F3F3F3;
  line-height: 55px;
  padding: 0 4px 0 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  cursor: pointer;
  outline: none;
  -webkit-tap-highlight-color: transparent;
  -webkit-transition: background-color 0.2s ease-out;
  transition: background-color 0.2s ease-out;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -moz-user-select: none;
}

.c13 i:first-of-type {
  font-size: 1rem;
  margin-left: 5px;
  margin-right: 10px;
  color: #D48C24;
}

.c13:hover {
  background-color: #FDF3E5;
}

.c13 i:last-of-type {
  position: absolute;
  z-index: 1;
  top: 14px;
  right: 7px;
  font-size: 1.7rem;
}

.c14 {
  visibility: hidden;
  opacity: 0;
  overflow: auto;
  -webkit-transition: opacity 0.2s ease 0s;
  transition: opacity 0.2s ease 0s;
  position: absolute;
  box-sizing: border-box;
  background-color: #F3F3F3;
  z-index: 2;
  top: 0;
  left: 0;
  padding: 20px 10px 10px;
  width: 100%;
  height: 100%;
  max-height: 100vh;
  border: 1px solid #F3F3F3;
  border-radius: 4px;
  border-top: 0;
  border-bottom: 0;
  outline: none;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 10px;
  margin: 10px 0 15px;
}

.c17 {
  font-size: 1.3rem;
}

.c18 {
  cursor: pointer;
  line-height: 0;
  background-color: transparent;
  outline: none;
  -webkit-tap-highlight-color: transparent;
  margin-top: 4px;
}

.c18 i {
  font-size: 1.1rem;
  font-weight: 600;
}

.c18 i {
  color: #585858;
  font-size: 1.5rem;
}

.c19 {
  width: 100%;
  height: 55px;
  border-bottom: 1px solid #E5EAED;
  list-style: none;
  display: block;
  overflow: hidden;
  color: #2f3235;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding: 9px 11px;
  cursor: pointer;
  outline: none;
  -webkit-tap-highlight-color: transparent;
}

.c19:hover {
  background-color: #FDF3E5;
}

.c19:focus {
  background-color: #FDF3E5;
}

.c21 {
  width: 18px;
  margin-right: 12px;
}

.c20 {
  color: #2f3235;
  font-size: 0.9rem;
  font-weight: 400;
}

.c22 {
  width: 100%;
  height: 55px;
}

.c24 {
  width: 100%;
  height: 100%;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  text-transform: uppercase;
  padding: 0 15px;
  border-radius: 4px;
  cursor: pointer;
  outline: none;
  background-color: #FBA52B;
  color: #FFFFFF;
  font-weight: 600;
  -webkit-transition: background-color 0.2s ease-out;
  transition: background-color 0.2s ease-out;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c24 i {
  margin-right: 6px;
  font-size: 1.3em;
}

.c24:hover {
  background-color: #EC7E00;
}

.c24:focus {
  background-color: #EC7E00;
}

.c0 {
  width: 85%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  margin: 0 0 25px;
}

.c8 {
  background-color: #A7A7A7;
  width: 100%;
  height: 1px;
  margin: 0 0 25px 0;
  max-width: 550px;
}

.c9 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 22px;
}

.c9:first-child {
  margin-right: 2%;
}

.c11 {
  width: 100%;
}

.c11:first-of-type {
  margin-right: 2%;
  margin-bottom: 15px;
}

.c23 {
  width: 100%;
  margin-bottom: 10px;
}

@media only screen and (min-width:768px) {
  .c3 {
    max-width: 550px;
  }
}

@media only screen and (min-width:992px) {
  .c16 {
    font-size: 1.45rem;
  }
}

@media only screen and (min-width:768px) {
  .c10 {
    height: 55px;
    position: relative;
  }
}

@media only screen and (min-width:768px) {
  .c14 {
    max-height: 440px;
    height: auto;
    padding: 0;
    box-shadow: 0 2px 2px rgba(0,0,0,0.35);
    -webkit-transition: opacity 0.2s ease 0s,-webkit-transform 0.2s ease 0s;
    -webkit-transition: opacity 0.2s ease 0s,transform 0.2s ease 0s;
    transition: opacity 0.2s ease 0s,transform 0.2s ease 0s;
    -webkit-transform: translateY( 20px );
    -ms-transform: translateY( 20px );
    transform: translateY( 20px );
  }
}

@media only screen and (min-width:768px) {
  .c15 {
    display: none;
  }
}

@media only screen and (min-width:470px) {
  .c0 {
    width: 70%;
  }
}

@media only screen and (min-width:768px) {
  .c0 {
    width: 60%;
  }
}

@media only screen and (min-width:992px) {
  .c0 {
    width: 58rem;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media only screen and (min-width:1250px) {
  .c0 {
    width: 71rem;
  }
}

@media only screen and (min-width:992px) {
  .c2 {
    margin: 0 2% 0 0;
    width: 39%;
  }
}

@media only screen and (min-width:1200px) {
  .c2 {
    width: 44%;
  }
}

@media only screen and (min-width:992px) {
  .c8 {
    width: 1px;
    height: 90%;
    margin: 0 2% 0 0;
  }
}

@media only screen and (min-width:768px) {
  .c9 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media only screen and (min-width:992px) {
  .c9 {
    width: 47%;
    -webkit-box-pack: space-around;
    -webkit-justify-content: space-around;
    -ms-flex-pack: space-around;
    justify-content: space-around;
    margin-bottom: 0;
  }

  .c9:first-child {
    margin-right: 0;
  }
}

@media only screen and (min-width:768px) {
  .c11 {
    max-width: 270px;
  }

  .c11:first-of-type {
    margin-bottom: 0;
  }
}

@media only screen and (min-width:992px) {
  .c11 {
    margin-right: 2%;
    width: 48%;
  }
}

@media only screen and (min-width:768px) {
  .c23 {
    width: 300px;
    margin-bottom: 0;
  }
}

@media only screen and (min-width:992px) {
  .c23 {
    width: 10%;
  }
}

<div
  class="c0"
>
  <div
    class="c1 c2"
    data-testid="autocomplete"
  >
    <div
      class="c3 c4"
      data-testid="input-wrapper"
    >
      <i
        class="material-icons"
        data-testid="input-icon"
      >
        search
      </i>
      <input
        class="c5"
        placeholder="Search for locals..."
        type="text"
        value=""
      />
    </div>
    <div
      class="c6"
      data-testid="listbox-wrapper"
    >
      <ul
        class="c7"
        role="listbox"
      />
    </div>
  </div>
  <span
    class="c8"
  />
  <div
    class="c9"
  >
    <div
      class="c10 c11"
      data-testid="dropdown"
    >
      <div
        class="c12"
      >
        <button
          class="c13"
          type="button"
        >
          <i
            class="material-icons"
            data-testid="label-icon"
          >
            near_me
          </i>
          <span>
            Select any location
          </span>
          <i
            class="material-icons"
          >
            arrow_drop_down
          </i>
        </button>
      </div>
      <div
        class="c14"
        data-testid="dropdown-list"
        role="listbox"
        tabindex="0"
      >
        <div
          class="c15"
        >
          <h4
            class="c16 c17"
          >
            Select any location
          </h4>
          <button
            class="c18"
            type="button"
          >
            <i
              class="material-icons"
            >
              close
            </i>
          </button>
        </div>
        <div
          class="c19"
          role="option"
        >
          <p
            class="c20"
          >
            Dublin
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <p
            class="c20"
          >
            South City West
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <p
            class="c20"
          >
            Temple Bar
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <p
            class="c20"
          >
            Clondalkin
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <p
            class="c20"
          >
            Swords
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <p
            class="c20"
          >
            Tallaght
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <p
            class="c20"
          >
            Blanchardstown
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <p
            class="c20"
          >
            South City East
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <p
            class="c20"
          >
            Dundrum
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <p
            class="c20"
          >
            Finglas
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <p
            class="c20"
          >
            Ballyfermot
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <p
            class="c20"
          >
            Santry
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <p
            class="c20"
          >
            Lucan
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <p
            class="c20"
          >
            Artane
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <p
            class="c20"
          >
            Malahide
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <p
            class="c20"
          >
            Dun Laoghaire
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <p
            class="c20"
          >
            Walkinstown
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <p
            class="c20"
          >
            Blackrock
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <p
            class="c20"
          >
            Palmerstown
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <p
            class="c20"
          >
            Rathmines
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <p
            class="c20"
          >
            Balbriggan
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <p
            class="c20"
          >
            Ballybrack
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <p
            class="c20"
          >
            Smithfield
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <p
            class="c20"
          >
            Crumlin
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <p
            class="c20"
          >
            Castleknock
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <p
            class="c20"
          >
            Howth
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <p
            class="c20"
          >
            Cabra
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <p
            class="c20"
          >
            Whitehall
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <p
            class="c20"
          >
            Portmarnock
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <p
            class="c20"
          >
            Kimmage
          </p>
        </div>
      </div>
    </div>
    <div
      class="c10 c11"
      data-testid="dropdown"
    >
      <div
        class="c12"
      >
        <button
          class="c13"
          type="button"
        >
          <i
            class="material-icons"
            data-testid="label-icon"
          >
            restaurant
          </i>
          <span>
            Select any cuisine
          </span>
          <i
            class="material-icons"
          >
            arrow_drop_down
          </i>
        </button>
      </div>
      <div
        class="c14"
        data-testid="dropdown-list"
        role="listbox"
        tabindex="0"
      >
        <div
          class="c15"
        >
          <h4
            class="c16 c17"
          >
            Select any cuisine
          </h4>
          <button
            class="c18"
            type="button"
          >
            <i
              class="material-icons"
            >
              close
            </i>
          </button>
        </div>
        <div
          class="c19"
          role="option"
        >
          <img
            alt="Irish"
            class="c21"
            src="http://localhost:3000/static/images/stew.svg"
          />
          <p
            class="c20"
          >
            Irish
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <img
            alt="Italian"
            class="c21"
            src="http://localhost:3000/static/images/pizza.svg"
          />
          <p
            class="c20"
          >
            Italian
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <img
            alt="Mediterranean"
            class="c21"
            src="http://localhost:3000/static/images/bruschetta.svg"
          />
          <p
            class="c20"
          >
            Mediterranean
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <img
            alt="Asian"
            class="c21"
            src="http://localhost:3000/static/images/rice.svg"
          />
          <p
            class="c20"
          >
            Asian
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <img
            alt="Healthy Food"
            class="c21"
            src="http://localhost:3000/static/images/healthy_food.svg"
          />
          <p
            class="c20"
          >
            Healthy Food
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <img
            alt="Indian"
            class="c21"
            src="http://localhost:3000/static/images/gulab.svg"
          />
          <p
            class="c20"
          >
            Indian
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <img
            alt="Mexican"
            class="c21"
            src="http://localhost:3000/static/images/taco.svg"
          />
          <p
            class="c20"
          >
            Mexican
          </p>
        </div>
        <div
          class="c19"
          role="option"
        >
          <img
            alt="Fast Food"
            class="c21"
            src="http://localhost:3000/static/images/fries.svg"
          />
          <p
            class="c20"
          >
            Fast Food
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="c22 c23"
  >
    <button
      class="c24"
      type="button"
    >
      Search
    </button>
  </div>
</div>
`;
